lib/route_types.txt

```diff
@@ -85,6 +85,7 @@ ZEBRA_ROUTE_BFD,        bfd,       bfdd,   '-', 0, 0, 0,     "BFD",           bf
 ZEBRA_ROUTE_OPENFABRIC, openfabric, fabricd,  'f', 1, 1, 1,  "OpenFabric",    fabricd
 ZEBRA_ROUTE_VRRP,       vrrp,      vrrpd,  '-', 0, 0, 0,     "VRRP",          vrrpd
 ZEBRA_ROUTE_NHG,        zebra,     none,   '-', 0, 0, 0,     "Nexthop Group", none
+ZEBRA_ROUTE_NEIGH,      neigh,     none,   '-', 0, 0, 0,     "Neigh Entry",   none
```

Comment:
gosh, Phillipe: this doesn't like a good idea. the net effect of all the code you have that adds this route type is just to get `netlink_neigh_update_msg_encode()` to use `RTPROT_ZEBRA` - if I'm following the paths that get there. there must be an easier way to do that - when I made the comment, I imagined you'd just say "if proto is zero, use ZEBRA, otherwise use the convert function." The point of the comment was that the evpn path was passing zero.

Reply:
:) adding  `ZEBRA_ROUTE_NEIGH` does not cost a lot.
what you expect if I get it right, is to not have the unguarded debug trace.
I will do an alternate proposal, then.

Reply:
@mjstapp , please tell me if last commit would be ok ?